python/activator/middleware_interface.py

```diff
@@ -176,42 +184,122 @@ def prep_butler(self, visit: Visit) -> None:
+            Center of the region to find refcat shards in.
+        radius : `lst.geom.Angle`
+            Radius to search for refcat shards in.
+        """
+        htmIndexer = HtmIndexer(depth=7)
+        shards = htmIndexer.getShardIds(center, radius+self.padding)
+        # getShardIds returns a tuple, the first item is the ids list.
+        htm_where = f"htm7 in ({','.join(str(x) for x in shards[0])})"
+        # Get shards from all refcats that overlap this detector.
+        # TODO: `...` doesn't work for this queryDatasets call
+        # currently, and we can't queryDatasetTypes in just the refcats
+        # collection, so we have to specify a list here. Replace this
+        # with another solution ASAP.
+        possible_refcats = ["gaia", "panstarrs", "gaia_dr2_20200414", "ps1_pv3_3pi_20170110"]
+        export.saveDatasets(self.central_butler.registry.queryDatasets(possible_refcats,
+                                                                       collections="refcats",
+                                                                       where=htm_where,
+                                                                       findFirst=True))
+
+    def _export_skymap_and_templates(self, export, center, detector, wcs):
+        """Export the skymap and templates for this visit from the central
+        butler.
+
+        Parameters
+        ----------
+        export : `Iterator[RepoExportContext]`
+            Export context manager.
+        center : `lsst.geom.SpherePoint`
+            Center of the region to load the skyamp tract/patches for.
+        detector : `lsst.afw.cameraGeom.Detector`
+            Detector we are loading data for.
+        wcs : `lsst.afw.geom.SkyWcs`
+            Rough WCS for the upcoming visit, to help finding patches.
+        """
+        export.saveDatasets(self.central_butler.registry.queryDatasets("skyMap",
+                                                                       collections="skymaps",
+                                                                       findFirst=True))
+        # Getting only one tract should be safe: we're getting the
+        # tract closest to this detector, so we should be well within
+        # the tract bbox.
+        skymap = self.central_butler.get("skyMap")
+        tract = skymap.findTract(center)
+        points = [center]
+        for corner in detector.getCorners(lsst.afw.cameraGeom.PIXELS):
+            points.append(wcs.pixelToSky(corner))
+        patches = tract.findPatchList(points)
+        template_where = f"patch in ({','.join(str(p.sequential_index) for p in patches)}) and tract={tract.tract_id}"
+        # TODO: do we need to have the coadd name used in the pipeline
+        # specified as a class kwarg, so that we only load one here?
+        # TODO: alternately, we need to extract it from the pipeline?
+        # TODO: alternately, can we just assume that there is exactly
+        # one coadd in the central butler?
```

Comment:
```suggestion
        # one coadd type in the central butler?
```

And I think the last is a safe assumption, at least for now. Only if the central repo started including DRP outputs would it be likely to be broken.